### ISSUE TITLE
Figure out what to do in tick when tween is completed via set_{elapsed,progress}

### ISSUE THREAD

**Issue by SUPERCILEX** [djeedai/bevy_tweening#43]
Currently we'll block you from updating the underlying structure if the tween is complete, but we should probably remove that optimization or keep track of whether or not the underlying was updated and perform the update even if the tween is completed.

**Comment by djeedai** [comment]
What does that mean exactly? Are you saying that when the tweenable completes, we can't use `tick()` anymore (fine, probably)? Or that the last tick that completes the tweenable does not apply its changes to the underlying object (component/asset), which would be a bug?

**Comment by SUPERCILEX** [comment]
> Are you saying that when the tweenable completes, we can't use tick() anymore (fine, probably)?

Yes, pretty much. This behavior has been around for a long as I can tell, but it feels wrong. Here's some pseudo code:

```
let transform;
let tween = from transform;
tween.set_elapsed MAX OR tween.set_progress 1. // Complete the tween somehow
tween.tick

// transform is not updated
```

If this is the desired behavior, we should document it. Though I feel like the transform should be updated.

**Comment by djeedai** [comment]
Ah right, in that case indeed it's bad the transform doesn't get updated
